From 4dfebd313fabd79aaf679f6b01581483ed323787 Mon Sep 17 00:00:00 2001
From: Max Ihlenfeldt <max@igalia.com>
Date: Tue, 13 Jun 2023 15:48:12 +0000
Subject: [PATCH] Avoid capturing structured bindings

Similar to https://crrev.com/c/4526976, this patch removes some uses of
structured bindings that lead to compile errors when using older
versions of clang (observed with clang 15).

Upstream-Status: Inappropriate [specific to older versions of clang]
Signed-off-by: Max Ihlenfeldt <max@igalia.com>
---
 .../service_worker/service_worker_context_wrapper.cc      | 3 ++-
 media/base/cdm_promise_adapter.cc                         | 8 ++++----
 .../core/layout/ng/grid/ng_grid_layout_algorithm.cc       | 8 ++++++--
 3 files changed, 12 insertions(+), 7 deletions(-)

diff --git a/content/browser/service_worker/service_worker_context_wrapper.cc b/content/browser/service_worker/service_worker_context_wrapper.cc
index d8d68ec..1c0777b 100644
--- a/content/browser/service_worker/service_worker_context_wrapper.cc
+++ b/content/browser/service_worker/service_worker_context_wrapper.cc
@@ -1443,9 +1443,10 @@ void ServiceWorkerContextWrapper::MaybeProcessPendingWarmUpRequest() {
   auto [document_url, key, callback] = std::move(*request);

   DCHECK(document_url.is_valid());
+  auto document_url_spec = document_url.spec();
   TRACE_EVENT1("ServiceWorker",
                "ServiceWorkerContextWrapper::MaybeProcessPendingWarmUpRequest",
-               "document_url", document_url.spec());
+               "document_url", document_url_spec);

   context_core_->registry()->FindRegistrationForClientUrl(
       ServiceWorkerRegistry::Purpose::kNotForNavigation,
diff --git a/media/base/cdm_promise_adapter.cc b/media/base/cdm_promise_adapter.cc
index f013809..144809b 100644
--- a/media/base/cdm_promise_adapter.cc
+++ b/media/base/cdm_promise_adapter.cc
@@ -94,12 +94,12 @@ void CdmPromiseAdapter::RejectPromise(uint32_t promise_id,
 void CdmPromiseAdapter::Clear(ClearReason reason) {
   // Reject all outstanding promises.
   DCHECK(thread_checker_.CalledOnValidThread());
-  for (auto& [promise_id, promise] : promises_) {
+  for (auto& promise : promises_) {
     TRACE_EVENT_NESTABLE_ASYNC_END1(
-        "media", "CdmPromise", TRACE_ID_WITH_SCOPE("CdmPromise", promise_id),
+        "media", "CdmPromise", TRACE_ID_WITH_SCOPE("CdmPromise", promise.first),
         "status", "cleared");
-    promise->reject(CdmPromise::Exception::INVALID_STATE_ERROR,
-                    ToSystemCode(reason), "Operation aborted.");
+    promise.second->reject(CdmPromise::Exception::INVALID_STATE_ERROR,
+                           ToSystemCode(reason), "Operation aborted.");
   }
   promises_.clear();
 }
diff --git a/third_party/blink/renderer/core/layout/ng/grid/ng_grid_layout_algorithm.cc b/third_party/blink/renderer/core/layout/ng/grid/ng_grid_layout_algorithm.cc
index 29d3b99..6af269e 100644
--- a/third_party/blink/renderer/core/layout/ng/grid/ng_grid_layout_algorithm.cc
+++ b/third_party/blink/renderer/core/layout/ng/grid/ng_grid_layout_algorithm.cc
@@ -3452,7 +3452,9 @@ void NGGridLayoutAlgorithm::PlaceGridItems(
   DCHECK(out_row_break_between);

   const auto& container_space = ConstraintSpace();
-  const auto& [grid_items, layout_data, tree_size] = sizing_tree.TreeRootData();
+  const auto& grid_items = sizing_tree.TreeRootData().grid_items;
+  const auto& layout_data = sizing_tree.TreeRootData().layout_data;
+  const auto& tree_size = sizing_tree.TreeRootData().subtree_size;

   const auto* cached_layout_subtree = container_space.GridLayoutSubtree();
   const auto container_writing_direction =
@@ -3616,7 +3618,9 @@ void NGGridLayoutAlgorithm::PlaceGridItemsForFragmentation(

   // TODO(ikilpatrick): Update |SetHasSeenAllChildren| and early exit if true.
   const auto& constraint_space = ConstraintSpace();
-  const auto& [grid_items, layout_data, tree_size] = sizing_tree.TreeRootData();
+  const auto& grid_items = sizing_tree.TreeRootData().grid_items;
+  const auto& layout_data = sizing_tree.TreeRootData().layout_data;
+  const auto& tree_size = sizing_tree.TreeRootData().subtree_size;

   const auto* cached_layout_subtree = constraint_space.GridLayoutSubtree();
   const auto container_writing_direction =
